impl/src/main/java/io/jsonwebtoken/impl/security/KeysBridge.java

```diff
@@ -36,6 +36,7 @@ public final class KeysBridge {
 
     private static final String SUNPKCS11_GENERIC_SECRET_CLASSNAME = "sun.security.pkcs11.P11Key$P11SecretKey";
     private static final String SUNPKCS11_GENERIC_SECRET_ALGNAME = "Generic Secret"; // https://github.com/openjdk/jdk/blob/4f90abaf17716493bad740dcef76d49f16d69379/src/jdk.crypto.cryptoki/share/classes/sun/security/pkcs11/P11KeyStore.java#L1292
+    private static final String GENERIC_SECRET_ALGNAME = "GenericSecret";
```

Comment:
@mnylen Can you add a comment here about where this string comes from.

You have a great description in the conversation, but adding it inline in the code will help future viewers 😄 

@lhazlewood any idea if this should be made more generic somehow?  I'm not sure how often we would come across this type of problem.  (but I'm guessing it was a PITA to troubleshoot?)

Reply:
I can add the comment. I also agree that it'd be great to be able to have this support for generic secrets a bit more generic, but not sure how to do that. As far as I'm aware, there's no standard for these HSM-backed JCE providers to follow.

One option of course would be to wrap the key with some kind of additional metadata (such as algorithm), and then unwrap it in the algorithms before passing it to the provider. There's existing `ProviderKey` already, but that is unwrapped way before the key hits the algorithm implementations.

Reply:
I can add the comment. I also agree that it'd be great to be able to have this support for generic secrets a bit more generic, but not sure how to do that. As far as I'm aware, there's no standard for these HSM-backed JCE providers to follow.

One option of course would be to wrap the key with some kind of additional metadata (such as algorithm), and then unwrap it in the algorithms before passing it to the provider. There's existing ProviderKey already, but that is unwrapped way before the key hits the algorithm implementations.

Or perhaps add a mechanism for registering the secret class names that should not be treated as strictly in validations.

Reply:
@bdemers @mnylen I'm wondering if this should be made to handle both names, something along the lines of:

```java
if (algName.contains("Generic") && algName.contains("Secret")) ....
```
or
```java
if (algName.startsWith("Generic") && algName.endsWith("Secret"))...
```
whichever is faster.  (I'm mostly sure the .startsWith/.endsWith approach is faster)

Reply:
I don't mind doing the change. Wonder if just startsWith("Generic") would be enough though?